baseapp/options.go

```diff
@@ -95,12 +95,22 @@ func (app *BaseApp) SetParamStore(ps ParamStore) {
 	app.paramStore = ps
 }
 
-// SetAppVersion sets the application's version string.
-func (app *BaseApp) SetAppVersion(v string) {
+// SetVersion sets the application's version string.
+func (app *BaseApp) SetVersion(v string) {
 	if app.sealed {
-		panic("SetAppVersion() on sealed BaseApp")
+		panic("SetVersion() on sealed BaseApp")
 	}
+	app.version = v
+}
+
+// ProtocolVersionManager exposes functionality to set
+// BaseApp's protocol version.
+type ProtocolVersionManager interface {
```

Comment:
IMHO given the method included a better name is `ProtocolVersionSetter`

Reply:
This sounds good to me. I will also change the upgrade keeper's field to match (from `protocolManager` to `protocolSetter`.

Reply:
`protocolSetter` sounds odd. When reading it means that it's to inject some protocol. 

How about `versionSetter`? `appVersionSetter`? Or maybe `pv`? We should also avoid repeating the type name in variable name.

Reply:
either `versionSetter` or `appVersionSetter` sounds good to me

Reply:
good point @robert-zaremba. Changed it to versionSetter. Sounds much clearer 👍🏻